### PR TITLE
Update filenames in file banners when moving a type to a new file.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/74703

### CODE REVIEW

File: src/Features/Core/Portable/AddFileBanner/AddFileBannerHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    /// Looks at <paramref name="banner"/> to see if it contains the name of <paramref name="sourceDocument"/>
+    /// in it.  If so, those names will be replaced with <paramref name="destinationFilePath"/>.
+    /// </summary>
+    private static ImmutableArray<SyntaxTrivia> UpdateEmbeddedFileNames(
+        Document sourceDocument,
+        string? destinationFilePath,
+        ImmutableArray<SyntaxTrivia> banner,
+        Func<SyntaxTrivia, string, SyntaxTrivia> createTrivia)
+    {
+        var sourceName = IOUtilities.PerformIO(() => Path.GetFileName(sourceDocument.FilePath));
```
Comment (ToddGrun) [inline src/Features/Core/Portable/AddFileBanner/AddFileBannerHelpers.cs:54]: >Path.GetFileName [](http://example.com/codeflow?start=53&length=16)

didn't change in this PR, but I think Path.GetFileName doesn't do any IO, and won't throw those exceptions.

File: src/Features/Core/Portable/AddFileBanner/AddFileBannerHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        var newRoot = destinationRoot.ReplaceToken(
+            firstToken,
+            firstToken.WithLeadingTrivia(
+                sourceBanner.Concat(firstToken.LeadingTrivia.Skip(destinationBanner.Length))));
+        return destinationDocument.WithSyntaxRoot(newRoot);
+    }
+
+    /// <summary>
+    /// Looks at <paramref name="banner"/> to see if it contains the name of <paramref name="sourceDocument"/>
+    /// in it.  If so, those names will be replaced with <paramref name="destinationFilePath"/>.
```
Comment (ToddGrun) [inline src/Features/Core/Portable/AddFileBanner/AddFileBannerHelpers.cs:46]: >with [](http://example.com/codeflow?start=52&length=4)

nit: with -> "with the file name from "

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Features/Core/Portable/AddFileBanner/AddFileBannerHelpers.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/AddFileBanner/AddFileBannerHelpers.cs:54]: I'll do a pass later to address these

File: src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs [hunk 3]
```diff
@@ -146,7 +145,10 @@ private async Task<Document> AddNewDocumentWithSingleTypeDeclarationAsync(Docume
             // get the updated document, give it the minimal set of imports that the type
             // inside it needs.
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs:146]: Yeah. Seems incorrect.  Was already like that, so I'll move to another pr in this space to cleanup.

**Pull request merged**